CORE-791: Treat Arrow Functions As Primary Expressions

chrisnevers opened the pull request.

chrisnevers committed:
ES7 grammar. Handle lambdas as primary expression to avoid unwanted comma expressions

chrisnevers committed:
Incr version

chrisnevers committed:
Clean up

jeapostrophe commented:
I don't understand what this PR is. Did you write it or is it a merge from the main repo? If we merge this, does it affect Reach at all, or do we need to also update the reference there?

chrisnevers commented:
I wrote it & if we merge this, it won't affect Reach until we update the commit reference in `stack.yaml`

chrisnevers commented:
I based this off a PR that was previously merged and removed from our fork where function bodies are blocks/expressions instead of statements per the [ecma specification](https://tc39.es/ecma262/#prod-ConciseBody). However, I fixed the grammar so the arrow functions can be in object values and function arguments
